src/Libraries/Microsoft.Extensions.AI.AzureAIInference/AzureAIInferenceEmbeddingGenerator.cs

```diff
@@ -164,16 +164,35 @@ static void ThrowInvalidData() =>
     }
 
     /// <summary>Converts an extensions options instance to an Azure.AI.Inference options instance.</summary>
-    private EmbeddingsOptions ToAzureAIOptions(IEnumerable<string> inputs, EmbeddingGenerationOptions? options, EmbeddingEncodingFormat format)
+    private EmbeddingsOptions ToAzureAIOptions(IEnumerable<string> inputs, EmbeddingGenerationOptions? options)
     {
-        EmbeddingsOptions result = new(inputs)
+        if (options is null)
         {
-            Dimensions = options?.Dimensions ?? _dimensions,
-            Model = options?.ModelId ?? _metadata.DefaultModelId,
-            EncodingFormat = format,
-        };
+            return new EmbeddingsOptions(inputs)
+            {
+                Dimensions = _dimensions,
+                Model = _metadata.DefaultModelId,
+                EncodingFormat = EmbeddingEncodingFormat.Base64,
+            };
+        }
+
+        if (options.RawRepresentationFactory?.Invoke(this) is EmbeddingsOptions result)
+        {
+            // Input doesn't have a setter, so use reflection to set it.
+            typeof(EmbeddingsOptions)
+                .GetField("<Input>k__BackingField", BindingFlags.NonPublic | BindingFlags.Instance)!
+                .SetValue(result, inputs.ToList());
```

Comment:
This is unfortunate. I think the right answer is going to be to just ignore RawRepresentationFactory in this implementation for now, i.e. revert this file. There's really nothing meaningful on the EmbeddingOptions that someone might provide, and something like EncodingFormat if set could actually mess up our parsing of the response.

Reply:
I choose this approach considering that we can get rid of reflection on https://github.com/jozkee/azure-sdk-for-net/pull/1.

Reply:
I see. That's reasonable, the trouble is if the dependency revs and changes this field, it'll break.

Can we keep this one as it was but do the non-reflection / internals access in your other branch where it's possible?

Reply:
OK, will do.

Reply:
> would it be appropriate to go with RawRepresentationFactory here or does the EncodingFormat concern is enough reason to revert?

We could write the code to successfully handle EmbeddingEncodingFormat.Float (and .Base64 is already handled), which should be trivially handled via deserialization. We would want to throw if it's set to other values, as they don't make sense with an `Embedding<float>` output. We can subsequently add overloads of `AsIEmbeddingGenerator` that produce other embedding types (e.g. `BinaryEmbedding`) if that's deemed valuable, but by far the most common use today is with float.

Reply:
Can we address all of this, including enabling RawRepresentationFactory in AI.Inference in a follow up PR to unblock this one?

Reply:
Yes

Reply:
Then, if there's no more feedback, please approve.